### ISSUE TITLE
Promote LoggerProvider from experimental to stable

### ISSUE THREAD

**Issue by CodeBlanch** [open-telemetry/opentelemetry-dotnet#5442]
`LoggerProvider` & `LoggerProviderBuilder` were added as part of the effort to implement the [Log Bridge API](https://github.com/open-telemetry/opentelemetry-dotnet/issues/4433).

Later things were split into two different diagnostics:

* [LoggerProvider - OTEL1000](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/diagnostics/experimental-apis/OTEL1000.md)
* [Log Bridge API - OTEL1001](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/diagnostics/experimental-apis/OTEL1001.md)

This issue is to propose promoting `LoggerProvider` (OTEL1000) to a public API for 1.9.0.

Why do that?

In short, #4014. We don't have the same DI surface in logging that we do for metrics & tracing. `LoggerProvider` & `LoggerProviderBuilder` give us parity in logging with what we have for metrics & tracing.
